lib/Socket.js

```diff
@@ -61,7 +60,8 @@ class Socket extends EventEmitter {
       args = args || {};
       let address = args.address || 'localhost';
 
-      if (!url.parse(address).port) {
+      // If no port was provided, add the default port.
+      if (!/(?::\d{2,4})/.test(address)) {
```

Comment:
👍 Yeah this is probably sufficient for input validation, means the builds can ditch `url` as well.